.github/workflows/publish-docker-image.yml

```diff
@@ -2,61 +2,151 @@ name: Publish Docker Image
 
 on:
   workflow_dispatch:
+  push:
+    branches:
+      - "main"
   release:
-    types: [prereleased]
+    types: [prereleased,released]
+
+env:
+  REGISTRY_IMAGE: daeuniverse/daed
+
 
 jobs:
+
+  build-web:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        # with:
+        #   submodules: 'recursive'
```

Comment:
We did it on purpose, as those source code files under submodules are also used by linux distro packages, such as archlinux and gentoo. For example, the packaging process of gentoo is done inside a isolated environment, which means we can't do git clone as we desire.

Reply:
Yeah, they are requested by package maintainers